apt-hook/json-hook.cc

```diff
@@ -0,0 +1,302 @@
+            if (!has_key) {
+                continue;
+            }
+
+            json_object *install;
+            has_key = json_object_object_get_ex(versions, "install", &install);
+            if (!has_key) {
+                continue;
+            }
+
+            if (version_from_origin_and_archive_ends_with(install, "UbuntuESMApps", "-apps-security")) {
+                result.esm_apps += 1;
+            } else if (version_from_origin_and_archive_ends_with(install, "UbuntuESM", "-infra-security")) {
+                result.esm_infra += 1;
+            } else if (version_from_origin_and_archive_ends_with(install, "Ubuntu", "-security")) {
+                result.standard += 1;
+            }
+        }
+    }
+
+    return true;
+}
+
+void print_counts(security_package_counts &counts) {
+    std::vector<std::string> count_msgs;
+    bool wrote_security = false;
+
+    if (counts.standard + counts.esm_infra + counts.esm_apps == 0) {
+        return;
+    }
+
+    if (counts.standard > 0) {
+        std::stringstream ss;
+        ss << counts.standard << " standard security update";
+        if (counts.standard != 1) {
+            ss << "s";
+        }
+        count_msgs.push_back(ss.str());
+        wrote_security = true;
+    }
+    if (counts.esm_infra > 0) {
+        std::stringstream ss;
+        ss << counts.esm_infra << " esm-infra ";
+        if (!wrote_security) {
+            ss << "security ";
+        }
+        ss << "update";
+        if (counts.esm_infra != 1) {
+            ss << "s";
+        }
+        count_msgs.push_back(ss.str());
+        wrote_security = true;
+    }
+    if (counts.esm_apps > 0) {
+        std::stringstream ss;
+        ss << counts.esm_apps << " esm-apps ";
+        if (!wrote_security) {
+            ss << "security ";
+        }
+        ss << "update";
+        if (counts.esm_apps != 1) {
+            ss << "s";
+        }
+        count_msgs.push_back(ss.str());
+    }
+
+    for (uint i = 0; i < count_msgs.size(); i++) {
+        if (count_msgs.size() == 3) {
+            if (i == 1) {
+                std::cout << ", ";
+            } else if (i == 2) {
+                std::cout << ", and ";
+            }
+        } else if (count_msgs.size() == 2) {
+            if (i == 1) {
+                std::cout << " and ";
+            }
+        }
+        std::cout << count_msgs[i];
+    }
+    std::cout << std::endl;
+}
+
+int main(int argc, char *argv[])
+{
+    (void) argc;
+    (void) argv;
+
+    char *fd_c_str = getenv("APT_HOOK_SOCKET");
+    if (fd_c_str == NULL) {
+       return 0;
+    }
+    std::string fd_str(fd_c_str);
+    if (fd_str == "") {
+       return 0;
+    }
+    int fd = stoi(fd_str);
+    __gnu_cxx::stdio_filebuf<char> apt_socket_in(fd, std::ios::in);
+    __gnu_cxx::stdio_filebuf<char> apt_socket_out(fd, std::ios::out);
+    std::istream socket_in(&apt_socket_in);
+    std::ostream socket_out(&apt_socket_out);
+
+
```

Comment:
We can remove the extra line here

Reply:
I added a couple extra lines to separate chunks of logic inside `main` (including this one). I can remove them if you'd like though

Reply:
I would remove it, but no worries if you want to keep it

Reply:
Removed!